statsmodels/base/optimizer.py

```diff
@@ -156,11 +155,14 @@ def _fit(self, objective, gradient, start_params, fargs, kwargs,
         extra_fit_funcs = kwargs.setdefault('extra_fit_funcs', dict())
 
         methods = ['newton', 'nm', 'bfgs', 'lbfgs', 'powell', 'cg', 'ncg',
-                'basinhopping']
+                'basinhopping', 'dogleg']
```

Comment:
I definitely like that idea more, but what about existing methods?  Should I keep them as is or refactor to make use of `_fit_minimize`?  

On a somewhat related note, I've noticed that `newton` method has it's own implementation entirely. Is there a specific edge case that `scipy.optimize.newton` doesn't cover?